program/tests/stake_instruction.rs

```diff
@@ -0,0 +1,7076 @@
+    AccountSharedData::new(0, 0, &id())
+}
+
+fn invalid_stake_state_pubkey() -> Pubkey {
+    Pubkey::from_str("BadStake11111111111111111111111111111111111").unwrap()
+}
+
+fn invalid_vote_state_pubkey() -> Pubkey {
+    Pubkey::from_str("BadVote111111111111111111111111111111111111").unwrap()
+}
+
+fn spoofed_stake_state_pubkey() -> Pubkey {
+    Pubkey::from_str("SpoofedStake1111111111111111111111111111111").unwrap()
+}
+
+fn spoofed_stake_program_id() -> Pubkey {
+    Pubkey::from_str("Spoofed111111111111111111111111111111111111").unwrap()
+}
+
+fn process_instruction(
+    mollusk: &Mollusk,
+    instruction_data: &[u8],
+    mut transaction_accounts: Vec<(Pubkey, AccountSharedData)>,
+    instruction_accounts: Vec<AccountMeta>,
+    expected_result: Result<(), ProgramError>,
+) -> Vec<AccountSharedData> {
+    for ixn_key in instruction_accounts.iter().map(|meta| meta.pubkey) {
+        if !transaction_accounts
+            .iter()
+            .any(|(txn_key, _)| *txn_key == ixn_key)
+        {
+            transaction_accounts.push((ixn_key, AccountSharedData::default()));
+        }
+    }
+
+    let instruction = Instruction {
+        program_id: id(),
+        accounts: instruction_accounts,
+        data: instruction_data.to_vec(),
+    };
+
+    let check = match expected_result {
+        Ok(()) => Check::success(),
+        Err(e) => Check::err(e),
+    };
+
+    let result =
+        mollusk.process_and_validate_instruction(&instruction, &transaction_accounts, &[check]);
+
+    result
+        .resulting_accounts
+        .into_iter()
+        .map(|(_, account)| account)
+        .collect()
+}
+
+fn get_default_transaction_accounts(instruction: &Instruction) -> Vec<(Pubkey, AccountSharedData)> {
+    let mut pubkeys: HashSet<Pubkey> = instruction
+        .accounts
+        .iter()
+        .map(|meta| meta.pubkey)
+        .collect();
+    pubkeys.insert(clock::id());
+    pubkeys.insert(epoch_schedule::id());
+    pubkeys.insert(stake_history::id());
+    #[allow(deprecated)]
+    pubkeys
+        .iter()
+        .map(|pubkey| {
+            (
+                *pubkey,
+                if clock::check_id(pubkey) {
+                    create_account_shared_data_for_test(&clock::Clock::default())
+                } else if rewards::check_id(pubkey) {
+                    create_account_shared_data_for_test(&rewards::Rewards::new(0.0))
+                } else if stake_history::check_id(pubkey) {
+                    create_account_shared_data_for_test(&StakeHistory::default())
+                } else if stake_config::check_id(pubkey) {
+                    config::create_account(0, &stake_config::Config::default())
+                } else if epoch_schedule::check_id(pubkey) {
+                    create_account_shared_data_for_test(&EpochSchedule::default())
+                } else if rent::check_id(pubkey) {
+                    create_account_shared_data_for_test(&Rent::default())
```

Comment:
It's still a bit annoying to do, but you do have this API for keyed sysvar accounts available from the library:
https://github.com/buffalojoec/mollusk/blob/0c7b0edfe5788e9a6892d2c4eb60d43390c7175f/harness/src/sysvar.rs#L101-L134